tokio-postgres/src/lib.rs

```diff
@@ -241,14 +244,23 @@ pub enum AsyncMessage {
 
 /// Message returned by the `SimpleQuery` stream.
 #[non_exhaustive]
-#[derive(Debug)]
 pub enum SimpleQueryMessage {
     /// A row of data.
     Row(SimpleQueryRow),
     /// A statement in the query has completed.
     ///
-    /// The number of rows modified or selected is returned.
-    CommandComplete(u64),
+    /// The full tag, along with the rows modified or selected are returned.
+    CommandComplete(CommandCompleteContents),
+}
+
+#[derive(Eq, PartialEq)]
+pub struct CommandCompleteContents {
+    /// Fields, if relevant for the CommandCompleteContents.
+    pub fields: Option<Arc<[OwnedField]>>,
+    /// The number of rows modified or selected.
+    pub involved_rows: u64,
```

Comment:
nit: I'd prefer just "rows" but up to you

Reply:
Actually, we don't even need this field since this information is in the tag. Maybe we could add a `pub fn rows()` which parses the number of rows from the tag instead of doing that parsing at construction.

Reply:
Rows is fine. I kept it because the data was in the original enum variant as parsed, and I figured if we ever get this merged into upstream that it might be a good idea for other consumers to be able to easily get this info. But yeah, we could also move it to a rows function.

Reply:
I think for simplicity sake actually I might like to keep this as a simple field for now if that's alright with you. I'll add a TODO to consider refactoring to a helper method